src/compiler/crystal/macros.cr

```diff
@@ -2357,6 +2375,35 @@ module Crystal::Macros
     end
   end
 
+  # A macro fresh variable.
+  #
+  # Every variable `node` is equivalent to:
+  #
+  # ```
+  # {{ "%#{name}".id }}{% if exps = node.exps %}{{ "{#{exps.splat}}".id }}{% end %}
+  # ```
+  class MacroVar < ASTNode
+    # Returns the name of the fresh variable.
+    def name : MacroId
+    end
+
+    # Returns the associated indices of the fresh variable.
+    def exps : ArrayLiteral
```

Comment:
**thought:** I'm wondering if we should rather spell out the name of this method.
This would be in line with `Yield#expressions` and `Expressions#expressions` which have the same semantics.

We might want to adjust the internal API as well, which is easy because it's private.
But we should be sure about which spelling to introduce in the compiler's public macro API.

I suppose there's not much gained from using abbreviation or not. Most important is consistency.

```suggestion
    def expressions : ArrayLiteral
```

Reply:
This is still pending btw